extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1090-lms-assignment-submissions.sql

```diff
@@ -0,0 +1,90 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE OR ALTER PROCEDURE [lms].[harmonize_assignment_submissions] AS
+BEGIN
+    SET NOCOUNT ON;
+
+	-- Load temporal table
+	SELECT
+		lmsSubmission.AssignmentSubmissionIdentifier,
+		EDFISTUDENT.StudentUSI,
+		lmsxAssignment.AssignmentIdentifier,
+		lmsxAssignment.SchoolId,
+		submissionStatusDescriptor.DescriptorId,
+		lmsSubmission.SubmissionDateTime,
+		lmsSubmission.EarnedPoints,
+		lmsSubmission.Grade,
+		lmsSubmission.CreateDate,
+		lmsSubmission.LastModifiedDate,
+		lmsSubmission.DeletedAt
+	INTO #ALL_SUBMISSIONS
+	FROM LMS.AssignmentSubmission lmsSubmission
+	INNER JOIN LMSX.Assignment lmsxAssignment
+		ON lmsSubmission.AssignmentIdentifier = lmsxAssignment.AssignmentIdentifier
+	INNER JOIN LMS.LMSUser lmsUser
+		ON lmsUser.LMSUserIdentifier = lmsSubmission.LMSUserIdentifier
+	INNER JOIN EDFI.Descriptor submissionStatusDescriptor
+		ON submissionStatusDescriptor.CodeValue = lmsSubmission.SubmissionStatus
+	INNER JOIN EDFI.Student EDFISTUDENT
+		ON EDFISTUDENT.Id = lmsUser.EdFiStudentId
+
+		INNER JOIN LMSX.SubmissionStatusDescriptor lmsxSubmissionStatus
+			ON submissionStatusDescriptor.DescriptorId = lmsxSubmissionStatus.SubmissionStatusDescriptorId
+
+	-- Insert new records
+	INSERT INTO LMSX.AssignmentSubmission(
+		[AssignmentSubmissionIdentifier],
+		[StudentUSI],
+		[AssignmentIdentifier],
+		[SchoolId],
+		[SubmissionStatusDescriptorId],
+		[SubmissionDateTime],
+		[EarnedPoints],
+		[Grade],
+		[CreateDate],
+		[LastModifiedDate]
+	)
+	SELECT
+		AssignmentSubmissionIdentifier,
+		StudentUSI,
+		AssignmentIdentifier,
+		SchoolId,
+		DescriptorId,
+		SubmissionDateTime,
+		EarnedPoints,
+		Grade,
+		CreateDate,
+		LastModifiedDate
+	FROM #ALL_SUBMISSIONS
+	WHERE
+		#ALL_SUBMISSIONS.AssignmentSubmissionIdentifier NOT IN
+			(SELECT AssignmentSubmission.AssignmentSubmissionIdentifier FROM LMSX.AssignmentSubmission)
+		AND #ALL_SUBMISSIONS.DeletedAt IS NULL
+
+
+	-- Update existing records
+	UPDATE LMSX.AssignmentSubmission SET
+		LMSX.AssignmentSubmission.SubmissionStatusDescriptorId = #ALL_SUBMISSIONS.DescriptorId,
+		LMSX.AssignmentSubmission.SubmissionDateTime = #ALL_SUBMISSIONS.SubmissionDateTime,
+		LMSX.AssignmentSubmission.EarnedPoints = #ALL_SUBMISSIONS.EarnedPoints,
+		LMSX.AssignmentSubmission.Grade = #ALL_SUBMISSIONS.Grade,
+		LMSX.AssignmentSubmission.LastModifiedDate = GETDATE()
```

Comment:
In the case of updates, it is the API itself - not the ODS database - that sets the modification date. Therefore, it _is_ appropriate to set this date in the stored procedure.

Reply:
Thanks for the clarification!